### DISCUSSION TITLE
YouTube Video

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#354] [Component suggestions]
<img width="480" alt="visuele weergave van een youtube video cmponent" src="https://github.com/user-attachments/assets/dda504d5-727a-4126-a8f7-1ee679daeaf6">

We willen graag meer leren over de **YouTube Video** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1730463300688269)

---

## Naam

YouTube Video

## Doel

YouTube-video met de bedieningselementen die YouTube zelf biedt.

## Ook bekend als 

Video

## Gerelateerde componenten

[Video](https://github.com/orgs/nl-design-system/discussions/299)

## Onderzoek
[Automatische ondertiteling van YouTube - Kennisbank DigiToegankelijk](https://kennisbank.digitoegankelijk.nl/automatische-ondertiteling-van-youtube/)
In dit artikel wordt uitgelegd dat YouTube’s automatische ondertiteling vaak niet voldoet aan WCAG-eisen voor doven en slechthorenden en daarom handmatige correctie en aanvulling nodig heeft om belangrijke informatie en geluiden nauwkeurig weer te geven.

[Sneltoetsen uitzetten in een videoplayer - Kennisbank DigiToegankelijk](https://kennisbank.digitoegankelijk.nl/sneltoetsen-uitzetten-in-een-videoplayer/)
In dit artikel wordt besproken dat sneltoetsen in videospelers voor digitale toegankelijkheid beter kunnen worden uitgeschakeld, om te voorkomen dat gebruikers met spraakbediening of toetsenbordnavigatie onbedoeld functies activeren.

[YouTube embed, WCAG compliant](https://www.200ok.nl/tips/youtube-embed/)
In dit artikel beschrijft Jules Ernst de toegankelijkheidsrichtlijnen voor het embedden van YouTube-video’s op websites. 

[Why videos on GOV.UK use the YouTube video player](https://accessibility.blog.gov.uk/2020/03/16/why-videos-on-gov-uk-use-the-youtube-video-player/)
Artikel waarin wordt uitgelegd waarom GOV.UK de YouTube Video player gebruikt en hoe ze deze toegankelijker maken. Omdat iframes lastig te navigeren zijn voor schermlezergebruikers, heeft GOV.UK bijvoorbeeld de videotitel en het woord “(video)” toegevoegd aan het `title-attribuut` van de iframe. Dit maakt meteen duidelijk dat er een video aanwezig is, wat een algemeen toegankelijkheidsprobleem bij videospelers oplost.

[Embedded YouTube and Google Maps – are they accessible? - Vision Australia](https://www.visionaustralia.org/business-consulting/digital-access/blog/embedded-youtube-and-google-maps)
Dit artikel bespreekt de toegankelijkheidsuitdagingen van het embedden van YouTube-video’s via iframes. Het richt zich specifiek op de beperkingen in aanpasbaarheid van deze iframes en onderzoekt hoe goed de YouTube-speler voldoet aan de WCAG 2.1 niveau AA-richtlijnen, waarbij inhoudelijke toegankelijkheidsaspecten zoals ondertiteling en audiobeschrijvingen buiten beschouwing worden gelaten. Enkele interessante inzichten:

- [Probleem 1: De naam van de link naar het kanaalprofiel is verwarrend](https://arc.net/l/quote/vnqaptjt). Gebruikers van schermlezers raken waarschijnlijk in de war door de profielafbeelding van het kanaal, omdat het een link is met een `aria-label-attribuut` met de waarde "Foto van". Deze link is niet erg logisch. Idealiter zou er "Vision Australia Channel" moeten staan ​​om het doel van de link duidelijk te maken.
- [Probleem 2: Kleurcontrast van de tekst](https://arc.net/l/quote/okriezvq). De videotitel en videoduur hebben een kleurcontrast van minder dan 4,5:1. Tekst zoals “Later bekijken” en “Delen” wordt ondersteund door een icoon met een contrastverhouding van meer dan 3:1, wat voldoet aan de eis voor niet-tekstcontrast (1.4.11).
- [Probleem 3: Button Volledig scherm heeft geen semantische naam](https://arc.net/l/quote/isvkwsmk). De Button 'Volledig scherm' heeft  geen semantische naam, ondanks dat andere knoppen in de YouTube-speler op de juiste manier zijn gelabeld.
- [Probleem 4: Sneltoetsen voor tekens met één toets](https://arc.net/l/quote/hukieuob). De YouTube-speler maakt gebruik van sneltoetsen voor tekens met één toets. Dit kan problematisch zijn voor de toegankelijkheid, omdat ondersteunende technologieën zoals schermlezers vaak gebruikmaken van dezelfde toetsaanslagen. Hierdoor kunnen gebruikers van schermlezers per ongeluk iets activeren dat verband houdt met de video terwijl ze hun schermlezer proberen te gebruiken, of andersom. Gelukkig biedt de Embedded YouTube-speler de optie om de toetsaanslagen met één toets uit te schakelen met "disablekb". 
- [Andere kleine problemen](https://arc.net/l/quote/wprkcool). Deze problemen gaan over de toegankelijkheid van abonnement-gerelateerde functies en videobediening op YouTube voor schermlezers en toetsenbordnavigatie.

**Comment by Robbert** [comment]
Relevante artikelen van de DigiToegankelijk kennisbank:

- [Automatische ondertiteling van YouTube](https://kennisbank.digitoegankelijk.nl/automatische-ondertiteling-van-youtube/)
- [Sneltoetsen uitzetten in een videoplayer](https://kennisbank.digitoegankelijk.nl/sneltoetsen-uitzetten-in-een-videoplayer/)

**Reply by Rozerinay** [reply]
Dank je wel voor je input! Ik heb de artikelen toegevoegd aan [Onderzoek](https://arc.net/l/quote/cjnhapmv).

**Comment by Rozerinay** [comment]
## Rotterdam

[Publieke website](https://www.rotterdam.nl/hecht-verbonden)

---

### Screenshots / Varianten

<img width="600" alt="image" src="https://github.com/user-attachments/assets/a02ff990-868c-4910-b155-7c561fcd0e31">

---

### Notities
Automatische ondertiteling staat aan.

**Comment by Rozerinay** [comment]
## Utrecht

[Publieke website](https://www.utrecht.nl/ondernemen/ondernemen-met-impact/impactondernemers-in-uw-buurt)

---

### Screenshots / Varianten

<img width="600" alt="image" src="https://github.com/user-attachments/assets/10379937-6971-4cbc-808d-73ac64c989cc">

---

### Notities
- Automatische ondertiteling staat niet aan. 
- Onder de video staat een link naar de video met audiodescriptie op YouTube.

**Comment by Rozerinay** [comment]
## Delft

[Publieke website](https://www.delft.nl/videos)

---

### Screenshots / Varianten

<img width="600" alt="image" src="https://github.com/user-attachments/assets/838faec3-00f5-4517-886c-1905e9ce0ec9">

---

### Notities
Automatische ondertiteling staat niet aan.

**Comment by Rozerinay** [comment]
## Belastingdienst

[Publieke website](https://www.belastingdienst.nl/wps/wcm/connect/nl/startende-ondernemer/)

---

### Screenshots / Varianten

<img width="793" alt="image" src="https://github.com/user-attachments/assets/4c2eeb75-b5f6-448b-8e70-c5bf06396db1">

---

### Notities
- Automatische ondertiteling staat niet aan. 
- Onder de video worden een uitleg van de inhoud, een downloadoptie en de uitgeschreven tekst aangeboden.

**Comment by Rozerinay** [comment]
## Westland

[Publieke website](https://www.gemeentewestland.nl/in-de-gemeente/over-westland/projecten/honselersdijk/flora-campus-westland/videos)

---

### Screenshots / Varianten

Toestemming vereist
<img width="500" alt="image" src="https://github.com/user-attachments/assets/9545714d-ab5e-46cb-becf-0394988078d7">

Video
<img width="685" alt="image" src="https://github.com/user-attachments/assets/cf450e43-1a8b-4780-8825-0d561c398838">

---

### Notities
- Toestemming vereist voordat video zichtbaar is.
- Vormgeving van deze YouTube player verschilt ten opzichte van andere YouTube players hierboven. Heeft minder bedieningselementen. 
- Automatische ondertiteling staat niet aan.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[YouTube Video op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=youtube+video)
